packages/cli/src/util/deploy/process-deployment.ts

```diff
@@ -114,32 +113,34 @@ export default async function processDeployment({
         const missingSize = missing
           .map((sha: string) => total.get(sha).data.length)
           .reduce((a: number, b: number) => a + b, 0);
+        const totalSizeHuman = bytes.format(missingSize, { decimalPlaces: 1 });
 
-        output.stopSpinner();
-        bar = new Progress(`${chalk.gray('>')} Upload [:bar] :percent :etas`, {
+        const progressOpts = {
           width: 20,
           complete: '=',
-          incomplete: '',
-          total: missingSize,
-          clear: true,
-        });
-
-        bar.tick(0);
+          incomplete: ' ',
+        };
 
         uploads.forEach((e: any) =>
           e.on('progress', () => {
-            if (!bar) return;
-
-            const totalBytesUploaded = uploads.reduce((acc: number, e: any) => {
+            const uploadedBytes = uploads.reduce((acc: number, e: any) => {
               return acc + e.bytesUploaded;
             }, 0);
-            // set the current progress bar value
-            bar.curr = totalBytesUploaded;
-            // trigger rendering
-            bar.tick(0);
 
-            if (bar.complete) {
+            if (uploadedBytes === missingSize) {
               output.spinner(deployingSpinnerVal, 0);
+            } else {
+              const bar = progress(uploadedBytes, missingSize, progressOpts);
```

Comment:
Since `bar` can be null now, should we add a debug log in that case?

Reply:
Ok added.